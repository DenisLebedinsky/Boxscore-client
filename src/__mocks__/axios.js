export default {
  get: params => {
    return Promise.resolve({
      data: {
        league: "NBA",
        away_team: {
          abbreviation: "OKC",
          first_name: "Oklahoma City"
        },
        home_team: {
          abbreviation: "MIA",
          first_name: "Miami"
        },
        away_period_scores: [26, 23, 22, 35],
        home_period_scores: [31, 28, 36, 26],
        event_information: {
          status: "completed"
        },
        extentions: {
          away_totals: {
            points: 106
          },
          home_totals: {
            points: 121
          },
          home_batter_totals: {
            hits: 13,
            batting_highlights: "5-29, HR, 3 R, 3 RBI, 2B"
          },
          away_batter_totals: {
            hits: 7,
            batting_highlights: "10-33, 4 R, 4 RBI, 2B, SB"
          }
        }
      }
    });
  }
};
