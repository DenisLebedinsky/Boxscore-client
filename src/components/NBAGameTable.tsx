import React, { useState, useEffect, ReactNode } from 'react';
import GameTable from './GameTable'

interface Ibatter_totals {
	batting_highlights: any;
	points: ReactNode;
	hits: ReactNode;
}

interface ITotalDetailsExtentions {
	away_totals: any;
	home_totals: any;
	away_batter_totals: Ibatter_totals;
	away_errors: string;
	home_batter_totals: Ibatter_totals;
}

interface ITotalDetails {
	extentions: ITotalDetailsExtentions;
}

interface IEventInformation {
	status: string;
}

interface ITeam {
	first_name: ReactNode;
	abbreviation: ReactNode;
}

interface IFooterGame {
	event_information: IEventInformation;
	away_team: ITeam;
	extentions: ITotalDetailsExtentions;
	home_team: ITeam;
}

interface IProps {
	getScoreTableUI(data: object): any,
	getTotalDetails(data: object): any,
	getFooterGame(data: object): any,
	fetchData(league: string): object;
	subscribeIo(league: string, setData: React.Dispatch<any>): object;
}

const NBAGameTable: React.FC<IProps> = props => {

	const league = 'NBA';
	const [data, setData] = useState();

	useEffect(() => {
		async function getData() {
			setData(await props.fetchData(league))
		}
		getData();
	}, [league])

	useEffect(() => {
		props.subscribeIo(league, setData);
	}, [league])

	const getTotalDetails = (data: ITotalDetails) => {
		if (!data) {
			return null;
		}
		return (
			<div className="totalDetails">
				<div className="boxscore__team__results boxscore__team__results--header">
					<span>Total</span>
				</div>
				<div className="boxscore__team__results">
					<span>{data.extentions.away_totals.points}</span>
				</div>
				<div className="boxscore__team__results">
					<span>{data.extentions.home_totals.points}</span>
				</div>
			</div>

		)
	};

	const getFooterGame = (data: IFooterGame) => {
		if (!data) {
			return null;
		}
		
		return (
			<div className="boxscore__details">
				<div className="boxscore__details__team boxscore__details__team--away">
					<p>
						<strong>{data.away_team.first_name}</strong><small>{data.away_team.abbreviation}</small>
					</p>
				</div>
				<div className="boxscore__details__info">
					<strong>{data.event_information.status}
						<br />
						4th {/*api response havn't this info*/}
					</strong>
				</div>
				<div className="boxscore__details__team boxscore__details__team--home">
					<p>
						<strong>{data.home_team.first_name}</strong><small>{data.home_team.abbreviation}</small>
					</p>
				</div>
			</div>

		)
	};

	return (
		<div>
			<div className="boxscore">
				<div className="tableScore">
					{props.getScoreTableUI(data)}
					{getTotalDetails(data)}
				</div>
				{getFooterGame(data)}
			</div>

		</div>
	)

}

export default GameTable(NBAGameTable);