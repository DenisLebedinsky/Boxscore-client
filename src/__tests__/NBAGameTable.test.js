import React from "react";
import NBAGameTable from "../components/NBAGameTable";
import {
  render,
  fireEvent,
  getByTestId,
  cleanup,
  waitForElement
} from "react-testing-library";
import axios from "axios";
import mockData from "../__mocks__/mockData";

describe("NBAGameTable", () => {
  afterEach(cleanup);

  axios.get = jest.fn();
  axios.get.mockResolvedValueOnce(mockData);

  const { container, getByText, getByTestId } = render(<NBAGameTable />);

  it("render data", async () => {
    const awayTotalPoints = await waitForElement(() =>
      getByTestId("awayTotalPoints")
    );

    const awayTeamNameAbr = await waitForElement(() =>
      getByTestId("awayTeamNameAbr")
    );

    const homeTeamName = await waitForElement(() =>
      getByTestId("homeTeamName")
    );

    expect(axios.get).toHaveBeenCalledTimes(1);
    expect(axios.get).toHaveBeenCalledWith("/data", {
      params: { league: "NBA" }
    });

    expect(container.firstChild).toMatchSnapshot();
  });
});
