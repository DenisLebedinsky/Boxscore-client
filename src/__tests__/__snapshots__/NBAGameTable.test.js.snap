// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NBAGameTable render data 1`] = `
<div
  class="boxscore"
>
  <div
    class="tableScore"
  >
    <div
      class="teamNames_scoreDetails"
    >
      <div
        class="teamNames"
      >
        <label
          class="head_lable"
        />
        <label
          data-testid="awayTeamNameAbr"
        >
          OKC
        </label>
        <label>
          MIA
        </label>
      </div>
      <div
        class="scoreDetails"
      >
        <div
          class="boxscore__team boxscore__team--header"
        >
          <div
            class="boxscore__team__units"
          >
            <span>
              1
            </span>
            <span>
              2
            </span>
            <span>
              3
            </span>
            <span>
              4
            </span>
          </div>
        </div>
        <div
          class="boxscore__team boxscore__team--away"
        >
          <div
            class="boxscore__team__units"
          >
            <span>
              26
            </span>
            <span>
              23
            </span>
            <span>
              22
            </span>
            <span>
              35
            </span>
          </div>
        </div>
        <div
          class="boxscore__team boxscore__team--home"
        >
          <div
            class="boxscore__team__units"
          >
            <span>
              31
            </span>
            <span>
              28
            </span>
            <span>
              36
            </span>
            <span>
              26
            </span>
          </div>
        </div>
      </div>
    </div>
    <div
      class="totalDetails"
    >
      <div
        class="boxscore__team__results boxscore__team__results--header"
      >
        <span
          class="totalCheck"
        >
          Total
        </span>
      </div>
      <div
        class="boxscore__team__results"
      >
        <span
          data-testid="awayTotalPoints"
        >
          106
        </span>
      </div>
      <div
        class="boxscore__team__results"
      >
        <span>
          121
        </span>
      </div>
    </div>
  </div>
  <div
    class="boxscore__details"
  >
    <div
      class="boxscore__details__team boxscore__details__team--away"
    >
      <p>
        <strong>
          Oklahoma City
        </strong>
        <small>
          OKC
        </small>
      </p>
    </div>
    <div
      class="boxscore__details__info"
    >
      <strong>
        completed
        <br />
        4th 
      </strong>
    </div>
    <div
      class="boxscore__details__team boxscore__details__team--home"
    >
      <p
        data-testid="homeTeamName"
      >
        <strong>
          Miami
        </strong>
        <small>
          MIA
        </small>
      </p>
    </div>
  </div>
</div>
`;
